cmd/openshift-install/targets.go

```diff
@@ -68,38 +68,28 @@ func newTargetsCmd() []*cobra.Command {
 
 func runTargetCmd(targets ...asset.WritableAsset) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
-		assetStore := &asset.StoreImpl{}
-		err := assetStore.Load(rootOpts.dir)
+		assetStore, err := asset.NewStore(rootOpts.dir)
 		if err != nil {
-			logrus.Errorf("Could not load assets from state file: %v", err)
+			return errors.Wrapf(err, "failed to create asset store")
 		}
+
 		for _, a := range targets {
-			err := assetStore.Fetch(a)
-			if err != nil {
+			if err := assetStore.Fetch(a); err != nil {
 				if exitError, ok := errors.Cause(err).(*exec.ExitError); ok && len(exitError.Stderr) > 0 {
 					logrus.Error(strings.Trim(string(exitError.Stderr), "\n"))
 				}
-				err = errors.Wrapf(err, "failed to generate %s", a.Name())
-			}
-
-			if err2 := asset.PersistToFile(a, rootOpts.dir); err2 != nil {
-				err2 = errors.Wrapf(err2, "failed to write asset (%s) to disk", a.Name())
-				if err != nil {
-					logrus.Error(err2)
-					return err
-				}
-				return err2
+				return errors.Wrapf(err, "failed to generate %s", a.Name())
 			}
+		}
 
-			if err != nil {
-				return err
-			}
+		if err := asset.PersistTargets(targets, rootOpts.dir); err != nil {
```

Comment:
https://github.com/openshift/installer/pull/454/commits/e7f24d6d2fdd5c9edf1dd2aa1e549530f232dbc8

Added the function that even if target generate fails, there might be partial state that might bee needed to be written to disk.